Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp

```diff
@@ -4180,11 +4143,13 @@ GstElement* MediaPlayerPrivateGStreamer::createVideoSink()
         return m_videoSink.get();
     }
 
-#if USE(GSTREAMER_HOLEPUNCH)
-    m_videoSink = createHolePunchVideoSink();
-    pushNextHolePunchBuffer();
-    return m_videoSink.get();
-#endif
+    if (isHolePunchRenderingEnabled()) {
+        m_videoSink = createHolePunchVideoSink();
+        if (m_videoSink) {
+            pushNextHolePunchBuffer();
+            return m_videoSink.get();
+        }
```

Comment:
It should return null here even if hole punching quirk did not return anything. This will preserve existing behavior we have, where we rely on playbin2 to find a suitable sink for us.